app/policies/subject_rule_effect_policy.rb

```diff
@@ -38,26 +21,38 @@ def edit?
   end
 
   def update?
-    # record is a subject_rule_effect from the controller
-    if credential.admin?
-      true
-    else
-      # this is a good place to check the defined subect set id in the
-      # rule effect belongs to the scoped credential project ids using the API client
+    return true if credential.admin?
+    return false unless valid_credentials?
+
+    if record.config.key?('subject_set_id')
+      subject_set = Effects.panoptes.subject_set(record.config['subject_set_id'])
+      raise ActiveRecord::RecordNotFound if subject_set.nil?
+
+      credential.project_ids.include?(subject_set['links']['project'])
+    elsif record.config.key?('collection_id')
+      collection = Effects.panoptes.collection(record.config['collection_id'])
+      raise ActiveRecord::RecordNotFound if collection.nil?
 
-      # temporarily disable the ability for users to create/update
-      # subject rule effects
-      false
+      credential.project_ids.include?(collection['links']['projects'].first)
+    else
+      true
     end
   end
 
   def destroy?
-    # record is a subject_rule_effect from the controller
     if credential.admin?
       true
     else
       subject_rule_project_id = record.subject_rule.workflow.project_id
       credential.project_ids.include?(subject_rule_project_id)
     end
   end
+
+  def validate_workflow
+    if credential.admin?
+      true
+    else
+      credential.project_ids.include?(record.project_id)
+    end
```

Comment:
inline with my other comments in the controller, i'd have policy actions like `index?` that check the credential.admin? status but then call this method to check the workflow is correct
```suggestion
    credential.project_ids.include?(record.project_id)
```
Each policy action method might look like this
```
  def index?
    if credential.admin?
      true
    else
      validate_workflow_record
    end
end 
```